app/services/builders/chartkick/project_distribution_data.rb

```diff
@@ -10,17 +10,19 @@ def call
       private
 
       def retrieve_records
-        merge_times
+        metric.retrieve_records(entity_id: @entity_id, time_range: @query[:value_timestamp])
       end
 
-      def merge_times
-        ::MergeTime.joins(pull_request: :project)
-                   .where(projects: { id: @entity_id })
-                   .where(pull_requests: { merged_at: @query[:value_timestamp] })
+      def metric_name
+        @query[:name]
+      end
+
+      def metric
+        @metric ||= ProjectDistributionDataMetrics.const_get(metric_name.to_s.camelize).new
```

Comment:
The cons I see with this dynamic approach is that it'd be easy to introduce a bug in the future. Another possible approach would be to have

```ruby
def metric
  @metric ||= case @query[:name]
              when :merge_time
                ProjectDistributionDataMetrics::MergeTime.new
              when :review_turnaround
                ProjectDistributionDataMetrics::ReviewTurnaround.new
              end
end
```

but it's true it's more verbose. So yeah.. I'm ok either way

Reply:
I'll keep it in mind.